File: java/src/main/java/javaprojet/Entity/BurgerCategorie.java
package javaprojet.Entity;

public class BurgerCategorie {
    private int id;
    private String nom;

    public BurgerCategorie(int id, String nom) {
        this.id = id;
        this.nom = nom;
    }

    public BurgerCategorie() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    @Override
    public String toString() {
        return "BurgerCategorie{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                '}';
    }
}

File: Symfony/BrasilBurgerGest/src/Entity/BurgerCategorie.php
<?php

namespace App\Entity;

use App\Repository\BurgerCategorieRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: BurgerCategorieRepository::class)]
class BurgerCategorie
{
    #[ORM\Id, ORM\GeneratedValue, ORM\Column(type:"bigint")]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $nom = null;

    /**
     * @var Collection<int, Burger>
     */
    #[ORM\OneToMany(mappedBy:"burgercategorie", targetEntity:Burger::class)]
    private Collection $burgers;

    public function __construct()
    {
        $this->burgers = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    /**
     * @return Collection<int, Burger>
     */
    public function getBurgers(): Collection
    {
        return $this->burgers;
    }

    public function addBurger(Burger $burger): static
    {
        if (!$this->burgers->contains($burger)) {
            $this->burgers->add($burger);
            $burger->setBurgercategorie($this);
        }

        return $this;
    }

    public function removeBurger(Burger $burger): static
    {
        if ($this->burgers->removeElement($burger)) {
            // set the owning side to null (unless already changed)
            if ($burger->getBurgercategorie() === $this) {
                $burger->setBurgercategorie(null);
            }
        }

        return $this;
    }
}
